src/opm/output/eclipse/Tables.cpp

```diff
@@ -1982,20 +1982,29 @@ namespace Opm {
 
     void Tables::addDensity( const DensityTable& density)
     {
-        if (density.size() > 0) {
-            const double default_value = -2e20;
-            const size_t num_columns = density[0].size;
-            std::vector<double> densityData( density.size() * num_columns , default_value);
-
-            this->m_tabdims[Ix::DensityNumTables] = density.size();
-            for (size_t table_num = 0; table_num < density.size(); table_num++) {
-                const auto& record = density[table_num];
-                densityData[ table_num * num_columns ]    = this->units.from_si( UnitSystem::measure::density , record.oil);
-                densityData[ table_num * num_columns + 1] = this->units.from_si( UnitSystem::measure::density , record.water);
-                densityData[ table_num * num_columns + 2] = this->units.from_si( UnitSystem::measure::density , record.gas);
+        if (density.size() == 0) { return; }
+
+        const double default_value = -2.0e20;
+        const auto nreg = density.size();
+        const auto nph  = density[0].size;
+
+        std::vector<double> densityData(nreg * nph, default_value);
+        {
+            const auto urho = UnitSystem::measure::density;
+
+            auto* rho_o = &densityData[0*nreg + 0]; // Oil   <-> Column 0
```

Comment:
The patch looks good, and will be merged. Just one question: why pointers and not iterators here?

Reply:
Didn't think about using iterators here, and pointers guarantee that we can use post increment without any additional cost.

Reply:
I can switch to using iterators if you'd like.

Reply:
It's ok as it is. I would have merged already if not for the unclear (to me) situation with the summary output changes.